spec/factories/course_assessment_tabs.rb

```diff
@@ -6,6 +6,8 @@
     end
     category do
       options = {}
+      # Run validation to make sure before_validation callbacks are called.
+      course.try(:valid?)
```

Comment:
@lowjoel I've updated this PR.

And this line is still needed because category's validation is actually ran before course's validation in this case.